fractal-theme/assets/js/src/search.js

```diff
@@ -58,40 +58,12 @@ function fractalSearch() {
           searchSuggestions.innerHTML = '<li><span class="search__no-results">No results</span></li>';
         }
       });
-
-      // Method for navigating with key up and key down in search suggestions
-      document.addEventListener('keydown', (e) => {
-        switch (e.key) {
-          case 'ArrowUp':
-            e.preventDefault();
-            if (document.activeElement === searchInput) {
-              break;
-            }
-            else if (!document.activeElement.parentNode.previousElementSibling) {
-              searchSuggestions.lastElementChild.firstElementChild.focus();
-            }
-            else {
-              document.activeElement.parentNode.previousElementSibling.firstElementChild.focus();
-            }
-            break;
-          case 'ArrowDown':
-            e.preventDefault();
-            if (document.activeElement === searchInput) {
-              searchSuggestions.firstElementChild.firstElementChild.focus();
-            }
-            else if (!document.activeElement.parentNode.nextElementSibling) {
-              searchSuggestions.firstElementChild.firstElementChild.focus();
-            }
-            else {
-              document.activeElement.parentNode.nextElementSibling.firstElementChild.focus();
-            }
-            break;
-        }
+      searchInput.addEventListener('focus', (e) => {
```

Comment:
I would prefer to add a single event listener on `keydown` and then check if search element has focus (optional).

It doesn't look like you're removing the event listeners again, so it's adding another event listener each time focus changes. This needs to be adressed :) I would suggest simply adding a event listener on `keydown`, checking that the search input has focus (and if so do the thing), and remove the event listener on `beforeunload`.

Reply:
On second thought: With this being plain JS, and the search thing being on all pages, maybe it's okay not to remove the eventListener - if you're only creating a single event listener, that is.

Reply:
> I would prefer to add a single event listener on `keydown` and then check if search element has focus (optional).

I startet out doing simething like this:
```
if (document.activeElement === searchInput) {
  initKeyboardNavigation();
}
```
But this does not work. I honestly don't know why. So I did this instead because that worked.
Even though i prefer to init the function if `SearchInout` has focus, I don't see whats wrong adding one more event listener. Is performance you are worried about or is it bad practice? Please teach me :)